app/grandchallenge/evaluation/forms.py

```diff
@@ -542,6 +542,37 @@ def clean_algorithm(self):
                 "complete."
             )
 
+        job_requirement_errors = []
+        phase = self.cleaned_data["phase"]
+        if (
+            algorithm.job_requires_memory_gb
+            > phase.algorithm_maximum_settable_memory_gb
+        ):
+            job_requirement_errors.append(
+                ValidationError(
+                    "The requested memory for this algorithm "
+                    f"({algorithm.job_requires_memory_gb}) is too high for this "
+                    "phase. The maximum allowed memory is "
+                    f"{phase.algorithm_maximum_settable_memory_gb} GB. "
+                    "Please adjust the setting on the algorithm."
+                )
+            )
+        if (
+            algorithm.job_requires_gpu_type
+            not in phase.algorithm_selectable_gpu_type_choices
+        ):
+            job_requirement_errors.append(
+                ValidationError(
+                    "The requested GPU type for this algorithm "
+                    f"({GPUTypeChoices(algorithm.job_requires_gpu_type).label}) is "
+                    "not allowed for this phase. Options are "
+                    f"{[GPUTypeChoices(c).label for c in phase.algorithm_selectable_gpu_type_choices]}. "
```

Comment:
Is using `.label` necessary in these places?

Reply:
I thought it would be nicest at first, because this is the representation that the user can know from the dropdown menu. But the labels are quite long to use them like this. I've changed it to using the name instead.